plugins/modules/ipaautomountlocation.py

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Authors:
+#   Chris Procter <cprocter@redhat.com>
+#
+# Copyright (C) 2021 Red Hat
+# see file 'COPYING' for use and warranty information
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.0",
+    "supported_by": "community",
+    "status": ["preview"],
+}
+
+
+DOCUMENTATION = '''
+---
+module: ipaautomountlocation
+author: chris procter
+short_description: Manage FreeIPA autommount locations
+description:
+- Add and delete an IPA automount location
+options:
+  ipaadmin_principal:
+    description: The admin principal
+    default: admin
+  ipaadmin_password:
+    description: The admin password
+    required: false
+  name:
+    description: The automount location to be managed
+    required: true
+    aliases: ["cn","location"]
+  state:
+    description: State to ensure
+    required: false
+    default: present
+    choices: ["present", "absent"]
+'''
+
+EXAMPLES = '''
+  - name: ensure a automount location named DMZ exists
+    ipaautomountlocation:
+      ipaadmin_password: SomeADMINpassword
+      name: DMZ
+      state: present
+
+  - name: ensure a automount location named DMZ is absent
+    ipaautomountlocation:
+      ipaadmin_password: SomeADMINpassword
+      name: DMZ
+      state: absent
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.ansible_freeipa_module import FreeIPABaseModule
+
+
+class AutomountLocation(FreeIPABaseModule):
+
+    ipa_param_mapping = {}
+
+    def get_location(self, location):
+        response = dict()
+        try:
+            response = self.api_command("automountlocation_show",
+                                        location,
+                                        {})
+        except Exception:
```

Comment:
Please, use as in the example given. If an exception other than `ipalib_erros.NotFound` is raised, it should not be hidden.